packages/inferno-animation/src/utils.ts

```diff
@@ -0,0 +1,206 @@
+    props.forEach((prop) => {
+      if (prop[0] === '-') {
+        let tmp = prop.split('-').splice(2).join('-')
+        if (props.indexOf(tmp) >= 0) {
+          nrofTransitions--
+        }
+      }
+    })
+    const animTimeout = dur.map((v, index) => parseFloat(v) + parseFloat(del[index])).reduce((prev, curr) => prev > curr ? prev : curr, 0)
+
+    nrofTransitions += dur.length
+    if (animTimeout > maxDuration) {
+      maxDuration = animTimeout
+    }
+  }
+
+  return {
+    nrofTransitions: nrofTransitions,
+    maxDuration: maxDuration
+  }
+}
+
+function whichTransitionEvent(){
+  var el = document.createElement('fakeelement');
+  var transitions = {
+    'transition':'transitionend',
+    'OTransition':'oTransitionEnd',
+    'MozTransition':'transitionend',
+    'WebkitTransition':'webkitTransitionEnd'
+  }
+
+  for(let t in transitions){
+      if( el.style[t] !== undefined ){
+          return transitions[t];
+      }
+  }
+}
+var transitionEndName: string;
+
+/**
+ * You need to pass the root element and ALL animated children that have transitions,
+ * if there are any,  so the timeout is set to the longest duration. Otherwise there
+ * will be animations that fail to complete before the timeout is triggered.
+ */
+export function registerTransitionListener(nodes: HTMLElement[], callback: Function, noTimeout: boolean = false) {
+  // I am doing this lazily because there where issues with document being undefined
+  // and checks seemed to go bust due to transpilation
+  if (!transitionEndName) {
+    transitionEndName = whichTransitionEvent()
+  }
+
+  if (!Array.isArray(nodes)) {
+    nodes = [nodes]
+  }
+  else {
+    // Make sure we don't have undefined nodes (happens when an animated el doesn't have children)
+    nodes = nodes.filter((node) => node)
+  }
+  const rootNode = nodes[0]
+
+  /**
+   * Here comes the transition event listener
+   */
+  let { nrofTransitions: nrofTransitionsLeft, maxDuration } = _getMaxTransitionDuration.apply(this, nodes)
+  let done = false
+
+  function onTransitionEnd (event) {
+    // Make sure this is an actual event
+    if (!event || done) {
+      return
+    }
+
+    if (!event.timeout) {
+      // Make sure it isn't a child that is triggering the event
+      var goAhead = false
+      for (var i=0; i < nodes.length; i++) {
+        if (event.target === nodes[i]) {
+          goAhead = true
+          break
+        }
+      }
+      if (!goAhead) return
+
+      // Wait for all transitions
+      if (--nrofTransitionsLeft > 0) {
+        return
+      }
+    }
+
+    // This is it...
+    done = true
+
+    /**
+     * Perform cleanup
+     */
+    rootNode.removeEventListener(transitionEndName, onTransitionEnd, false)
+    callback && callback()
+  }
+  rootNode.addEventListener(transitionEndName, onTransitionEnd, false)
+
+  // Fallback if transitionend fails
+  // This is disabled during debug so we can set breakpoints
+  if (!window.__DEBUG_ANIMATIONS__ && !noTimeout) {
```

Comment:
this debug thing could be moved to its own function and then prevent calling that funciton using: 
```
if (process.env.NODE_ENV !== 'production') {
}

```

to keep it inline with other infernojs debug messages

Reply:
@Havunen QUESTION: This would make debugging animations in production code using slowed down animations or by setting a breakpoint impossible. Do we want this? Should we provide an alternative solution to allow enabling this?

Reply:
using process.env.NODE_ENV production check would strip off this code from the production build. IMO debugging is for development bundles, at least this is how it is with all other code in infernojs

Reply:
FIXED in 538601bb0